core/src/autogluon/core/utils/feature_selection.py

```diff
@@ -400,8 +391,8 @@ def compute_next_candidate_given_fi(self, importance_df: pd.DataFrame, prune_thr
             return acceptance_candidates
 
         sample_weights = [1/i for i in range(1, len(below_threshold_rows)+1)] if weighted else None
-        while time_budget - (time.time() - time_start) > 0:
-            random_state = self.rng.integers(low=0, high=1e5)
+        for _ in range(50):
```

Comment:
This seems strange to me. Why is this a for loop (or a loop at all)?

Reply:
This is here to bypass cases where the sampled removal_candidates have already been pruned in previous fits but did not lead to validation set performance increase. This can happen a lot in low dimensional features when only a few features are below threshold. In those cases, we just sample again for up to 50 times and return the inputted features if no untried removal_candidates were found.

Reply:
why allow those invalid candidates to be chosen in the first place? Couldn't you just prune the list of features that are allowed to be picked instead of having this for loop?

Reply:
That would work for pruning single feature at a time but we are sampling multiple features, which would require us to keep a list whose size is combinatorial w.r.t. the prunable feature count.